### DISCUSSION TITLE
Add method to  < SpawnCharacter > to add custom characters to crewlist manager.

### DISCUSSION THREAD

**Discussion by phillipinman** [FakeFishGames/Barotrauma#14446] [Ideas and suggestions]
Hi,

While trying to create a custom creature and have the player persist as that creature in campaign *between* rounds I noticed there is no current way I am aware of that allows someone to be changed into say, a mudraptor, and continue to play as a mudraptor for the whole campaign (or until they die) between levels.

**My Issue**: Currently if a person is changed by < SpawnCharacter > into something else at the next start of the round they are respawned as a human.

**My Findings**: I was thinking as part of < SpawnCharacter > a new option could be added: AddToCrewList. From my testing the transformed player does get assigned their old TeamID already and looking at the C# code it seems that for them to persist between rounds as a changed Character they need to be apart of the crew manager. LUA can do this but it's overkill.

This would allow the player to persist as that custom character for the whole campaign or until they choose to re spawn.

As always, love your work and many thanks :)

**Comment by SomeRandomNoobKekeke** [comment]
How did you even managed to assign transformed character to Team1? There's no TeamID in CharacterSpawnInfo and NPCChangeTeamAction doesn't accept living players

**Reply by phillipinman** [reply]
@SomeRandomNoobKekeke You don't have to, the team is decided by the group in the Character definition, if you make it Group="human" much like pets they won't be hostile. Clown-O-Trauma uses this same mechanism to make zombies that dead players can control. :)

It is an issue when you give them human AI as they become hostile, so that is also a consideration for implementing a solution.

**Reply by SomeRandomNoobKekeke** [reply]
I just tried, it doesn't work in Clown-O-Trauma either, you're not on team1

**Reply by phillipinman** [reply]
You are correct, i just decompressed a save and it seems the Pet attribute is how it persists between rounds currently, So while a pet has no teams in the save file it does allow you to get a similar effect with setting the group to human so that they don't attack anyone.

The team1 issue should be fixed with the code in my other comment however, testing locally it seems to work fine :)

**Comment by phillipinman** [comment]
Based on a quick hack job the following works with the quirk that the original character will also re-spawn next round, but the custom character does persist between rounds:

StatusEffects.cs

            public bool AddToCrew { get; private set; }

            # Ommitted

                                    if (characterSpawnInfo.AddToCrew)
                                    {
                                            newCharacter.TeamID = CharacterTeamType.Team1;
                                            GameMain.GameSession.CrewManager.AddCharacter(newCharacter);
                                            return;
                                    }

This also fixes the fact that by default anything with a Human AI attribute (UseHumanAI="True") is hostile.

So the end result:

    We pass AddToCrew="True" to < SpawnCharacter >
    We create the new character and set the team to the same team as the crew. (If you add to crew first it will fail as they are hostile)
    We then add them to the crew manager.
    Success!

Edit: This only works in single player it seems. In multiplayer the effect does not persist between rounds.
